polkadot/xcm/src/v5/traits.rs

```diff
@@ -289,7 +295,8 @@ pub trait PreparedMessage {
 /// Type of XCM message executor.
 pub trait ExecuteXcm<Call> {
 	type Prepared: PreparedMessage;
-	fn prepare(message: Xcm<Call>) -> result::Result<Self::Prepared, Xcm<Call>>;
+	/// If it fails, returns the index of the problematic instruction.
+	fn prepare(message: Xcm<Call>) -> result::Result<Self::Prepared, u8>;
```

Comment:
What about introducing type alias for this instead of using `u8` in the apis?
```
pub type InstructionIndex = u8; // or something
```
Yes, it would be little bit verbose, but I think also little bit more readable.

(not blocking on this)